app/models/framework_question.rb

```diff
@@ -15,8 +15,38 @@ class FrameworkQuestion < VersionedModel
   belongs_to :framework
   has_many :dependents, class_name: 'FrameworkQuestion',
                         foreign_key: 'parent_id'
-
   belongs_to :parent, class_name: 'FrameworkQuestion', optional: true
+
   has_many :flags
   has_many :framework_responses
+
+  def build_responses(question: self, person_escort_record:, questions:)
```

Comment:
maybe this method can live in `person_escort_record` class to avoid passing around `questions`/`person_escort_record` 🤔

Reply:
That's worth looking at if it makes the implementation cleaner. This already violates the single responsibility slightly by building a bunch of responses inside the question class... not that it matters, so unless the PER class is already doing lots of things it wouldn't be any worse to move it. If we wanted to get really OOP and gang of five then it could be moved to a service so that it just does one thing but it feels like overkill to do that.